std/functional.d

```diff
@@ -104,31 +104,7 @@ the parameter or provide the symbol via the $(D parmName) argument.
 If $(D fun) is not a string, $(D unaryFun) aliases itself away to $(D fun).
 */
 
-template unaryFun(alias fun, string parmName = "a")
-{
-    static if (is(typeof(fun) : string))
-    {
-        static if (!fun._ctfeMatchUnary(parmName))
-        {
-            import std.traits, std.typecons, std.meta;
-            import std.algorithm, std.conv, std.exception, std.math, std.range, std.string;
-        }
-        auto unaryFun(ElementType)(auto ref ElementType __a)
-        {
-            mixin("alias " ~ parmName ~ " = __a ;");
-            return mixin(fun);
-        }
-    }
-    else static if (needOpCallAlias!fun)
-    {
-        // Issue 9906
-        alias unaryFun = fun.opCall;
-    }
-    else
-    {
-        alias unaryFun = fun;
-    }
-}
+alias unaryFun(alias fun, string parmName = "a") = naryFun!(fun, parmName);
```

Comment:
I do slightly prefer the original signatures. `unaryFun(alias fun, parmName = "a")` is more clear than `unaryFun(alias fun, parmName...) if (parmName.length <= 1)`.

If we turn `naryFunAliases` into a template, then multiple instantiations with the same `argc` won't have to re-evaluate. The problem is that `cache` would be re-evaluated for each different `argc`. You can un-nest it from `naryFunAliases` to avoid this, but I'd rather not create too many top-level private templates just to support this one. I thought maybe `static enum` would do the trick, but it doesn't look like that works. 

I'll just use the changed signatures for now.